bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java

```diff
@@ -839,6 +839,10 @@ public void checkpoint(Checkpoint checkpoint) throws IOException {
             throw e;
         } catch (RuntimeException e) {
             recordFailedEvent(dbLedgerStorageStats.getFlushStats(), startTime);
+            // not wrap IOException again
+            if (e.getCause() != null && e.getCause() instanceof IOException) {
```

Comment:
Does IOException throw at the first catch block?

Reply:
This IOException is wrapped by RuntimeException. not a directly IOException. The first catch block can not caught it.

Reply:
Could you share the detailed IOException wrapped by RuntimeException?

Reply:
Change the interface will be better:
```
    public interface EntryConsumer {
        void accept(long ledgerId, long entryId, ByteBuf entry);
    }
```

```
    public interface EntryConsumer {
        void accept(long ledgerId, long entryId, ByteBuf entry) throws IOException;
    }
```
So we needn't catch the inner exception. Throw the IOException directly.

Reply:
+1